src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/begrunnelser/FinnGyldigeBegrunnelserForPeriodeContext.kt

```diff
@@ -0,0 +1,213 @@
+
+            VedtakBegrunnelseType.FORTSATT_INNVILGET -> throw Feil("FORTSATT_INNVILGET skal være filtrert bort.")
+        }
+
+    private fun finnPersonerMedVilkårResultaterSomGjelderIPeriode(): Map<Person, List<VilkårResultat>> =
+
+        personResultater.tilFørskjøvetVilkårResultatTidslinjeMap(persongrunnlag)
+            .mapNotNull { (aktør, vilkårResultatTidslinjeForPerson) ->
+                val person =
+                    persongrunnlag.personer.find { it.aktør.aktivFødselsnummer() == aktør.aktivFødselsnummer() }
+                val forskøvedeVilkårResultaterMedSammeFom =
+                    vilkårResultatTidslinjeForPerson.tilPerioderIkkeNull().singleOrNull {
+                        it.fom == vedtaksperiode.fom
+                    }?.verdi
+                if (person != null && forskøvedeVilkårResultaterMedSammeFom != null) {
+                    Pair(person, forskøvedeVilkårResultaterMedSammeFom)
+                } else {
+                    null
+                }
+            }.toMap().filterValues { it.isNotEmpty() }
+
+    private fun finnPersonerMedVilkårResultaterSomGjelderRettFørPeriode(): Map<Person, List<VilkårResultat>> =
+        personResultater.tilFørskjøvetVilkårResultatTidslinjeMap(persongrunnlag).mapNotNull { (aktør, tidsjlinje) ->
+            val person = persongrunnlag.personer.find { it.aktør.aktivFødselsnummer() == aktør.aktivFødselsnummer() }
+            val forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode =
+                tidsjlinje.tilPerioderIkkeNull().singleOrNull {
+                    it.tom?.plusDays(1) == vedtaksperiode.fom
+                }?.verdi
+
+            if (person != null && forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode != null) {
+                Pair(person, forskøvedeVilkårResultaterSlutterDagenFørVedtaksperiode)
+            } else {
+                null
+            }
+        }.toMap().filterValues { it.isNotEmpty() }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPersonerUtenUtbetalingVedInnvilget(vedtakBegrunnelseType: VedtakBegrunnelseType) =
+        this.filterKeys {
+            if (vedtakBegrunnelseType == VedtakBegrunnelseType.INNVILGET) {
+                aktørIderMedUtbetaling.contains(it.aktør.aktørId) || it.type == PersonType.SØKER
+            } else {
+                true
+            }
+        }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPåVilkårType(vilkårTyperFraSanity: List<Vilkår>) =
+        this.mapValues { (_, vilkårResultaterForPerson) ->
+            vilkårResultaterForPerson.filter { vilkårTyperFraSanity.contains(it.vilkårType) }
+        }.filterValues { it.isNotEmpty() }
+
+    private fun Map<Person, List<VilkårResultat>>.filtrerPåTriggere(
+        triggereFraSanity: List<Trigger>
+    ) = this.filter { (person, vilkårResultaterForPerson) ->
+        Trigger.values().filter { it.erOppfylt(vilkårResultaterForPerson, person) } == triggereFraSanity
```

Comment:
Nå er denne strengere enn filtreringen på utdypende. 
```suggestion
        val oppfylteTriggereIBehandling = Trigger.values().filter { it.erOppfylt(vilkårResultaterForPerson, person) }
        triggereFraSanity.all { oppfylteTriggereIBehandling.contains(it) }
```

Reply:
Her blir det litt av det samme som for utdypendeVilkårsvurdering filtreringen. Må beholde streng sjekk for de 4 standard innvilgelses-begrunnelsene 🤔